src/EditorFeatures/CSharp/BlockCommentCompletion/BlockCommentCompletionCommandHandler.cs

```diff
@@ -0,0 +1,171 @@
+            }
+
+            var currentLineStartsWithBlockCommentStartString = snapshotLine.StartsWith(firstNonWhitespacePosition, "/*", false);
+            var currentLineStartsWithBlockCommentEndString = snapshotLine.StartsWith(firstNonWhitespacePosition, "*/", false);
+            var currentLineStartsWithBlockCommentMiddleString = snapshotLine.StartsWith(firstNonWhitespacePosition, "*", false);
+
+            if (!currentLineStartsWithBlockCommentStartString && !currentLineStartsWithBlockCommentMiddleString)
+            {
+                return null;
+            }
+
+            if (!IsCaretInsideBlockCommentSyntax(caretPosition))
+            {
+                return null;
+            }
+
+            if (currentLineStartsWithBlockCommentStartString)
+            {
+                if (BlockCommentEndsRightAfterCaret(caretPosition))
+                {
+                    //      /*|*/
+                    return " ";
+                }
+                else if (caretPosition == firstNonWhitespacePosition + 1)
+                {
+                    //      /|*
+                    return null; // The newline inserted could break the syntax in a way that this handler cannot fix, let's leave it.
+                }
+                else
+                {
+                    //      /*|
+                    return " *" + GetPaddingOrIndentation(snapshotLine, caretPosition, firstNonWhitespacePosition, "/*");
+                }
+            }
+
+            if (currentLineStartsWithBlockCommentEndString)
+            {
+                if (BlockCommentEndsRightAfterCaret(caretPosition))
+                {
+                    //      /*
+                    //      |*/
+                    return " ";
+                }
+                else if (caretPosition == firstNonWhitespacePosition + 1)
+                {
+                    //      *|/
+                    return "*";
+                }
+                else
+                {
+                    //      /*
+                    //   |   */
+                    return " * ";
+                }
+            }
+
+            if (currentLineStartsWithBlockCommentMiddleString)
+            {
+                if (BlockCommentEndsRightAfterCaret(caretPosition))
+                {
+                    //      *|*/
+                    return "";
+                }
+                else if (caretPosition > firstNonWhitespacePosition)
+                {
+                    //      *|
+                    return "*" + GetPaddingOrIndentation(snapshotLine, caretPosition, firstNonWhitespacePosition, "*");
+                }
+                else
+                {
+                    //      /*
+                    //   |   *
+                    return " * ";
+                }
+            }
+
+            return null;
+        }
+
+        private static bool BlockCommentEndsRightAfterCaret(SnapshotPoint caretPosition) => caretPosition.Snapshot.GetText(caretPosition, 2) == "*/";
```

Comment:
Ops, yes should have bound check here.

While adding tests for it, I discovered that `ITextSnapshotLine.StartsWith` always return false when the snapshot ends with value, I mean
`snapshotLine.StartsWith(0, "*", false) == false` when `snapshot.GetText() == "*"`

See https://github.com/dotnet/roslyn/blob/master/src/EditorFeatures/Text/Shared/Extensions/ITextSnapshotLineExtensions.cs#L151

Is that intentional?